### DISCUSSION TITLE
Come to say hi!

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#80] [Offtopic]
# Hi!

This is a discussion where you can shortly introduce yourself if you like to. You can either answer a few of these questions, or just write something freely. Even just a _Hi!_ without anything else in the message is welcome! 😎 

- Your real name (if you like)
- Where are you from?
- Some things of interest like occupation, education, hobbies, favorite colors, whatever.
- For what kind of things you use Obsidian?
- How did you find out about this plugin?
- How do you use this plugin, or plan to use in the future? (Shortly. For a bigger description, you can post to [Show and tell](https://github.com/Taitava/obsidian-shellcommands/discussions/categories/show-and-tell)).
---

I'll start. Hi, my name is Jarkko Linnanvirta. I'm from Finland, a country that is small by it's population, but large by it's area, located in North Europe, between Sweden and Russia.

My GitHub username _Taitava_ comes from my former one-man company named _IT-Palvelu Taitava_ (in English: IT Service Skillful), so *taitava* means _skillful_ in Finnish. The company does not exist anymore, but the name still lives here in GitHub. In [Obsidian forums](https://forum.obsidian.md/) I go by the name [Jare](https://forum.obsidian.md/u/Jare).

My occupation in the past has been a programmer (among other IT related occupations). Currently, I'm studying more programming in [Satakunta University of Applied Sciences](https://www.samk.fi/en/) and plan to do it for my work also in the future. It started as a hobby when I programmed small games as a teenager almost twenty years ago (a bit over ten years as an occupation).

In addition to being a programmer, I'd also like to be an author in the future and write novels. I'm also interested in finding ways to improve my mental health and developing my thinking and writing.

I'm using Obsidian for:
- Taking notes in school lectures and keeping track of task deadlines.
- Building a database of cat foods and whether my cat likes them or not.
- Building a database of all my devices that have a warranty, so that I can easily find out can I get them repared/replaced if they break down.
- Writing my novels.
- Writing a dream diary.

I store backups of all of these things in [Git](https://git-scm.com) repositories - not necessarily here in GitHub, some vaults' backups are stored in a portable hard drive instead of GitHub. I need to launch a Git client application many thimes during working with a vault (because Obsidian does not have a build-in Git client, like some programming editors have). For this reason, I needed to find a way to quickly launch external applications right from Obsidian with the vault's path passed to the external application (so that Git knows which Git repository to use). I did not find a plugin that would do this for me, so I created my own, and SC was born. The first version of SC did not have even vairables, but it worked well for my needs to just open up the Git client.

Huh, this introduction became longer than I expected.

**Comment by FelipeRearden** [comment]
As an INFJ it is a little bit difficult for me to talk about myself but I'm gonna try. 

Hi, my name is Felipe and I'm from Brazil. 

I have been using Obsidian since 2019 but I start a heavy usage in January of 2021. I'm using Obsidian for:

1 Building my own PKM and trying to transform into a IKE (integrated knowledge environment):

My dream is to use Obsidian as a lifetime project in a way that I could create an environment to concentrate every knowledge that is important to me that I don't want to forget and is created in a way that is easy to read again in the next years. 
- books
- articles 
- podcasts
- videos (YT, masterclass ...) 
- twitter
- light personal finances (spreadsheet) and graphs;
- ESL (english as a Second Language)
- Subjects that I want to learn in the future 

2 Applied the same principles described above to a vault for my daily job (IWE - integrated work environment) 

3 Use Obsidian as a Mental Health support 
- get rid of bad habits
- sustain good habits 
- habit tracker 
- mood tracker 
- time tracking 

4 Integrate Obsidian with other MacOS Apps and iOS Apps 

I'm using SC applied to all things I said above: since I start using SC and talking to @ I was able to do things that wasn't possible before. 

I have to say that SC is  the responsible for making my dream come true because step by step after every new update I'm getting closer to using Obsidian as my IKE and IWE
- I have to say that SC is changing my Obsidian-life :)

For my standards I think I said a lot :)

**Reply by Taitava** [reply]
Really interesting! I think I am bit by bit on my way on to do this kind of PKM things too.

> I have to say that SC is the responsible for making my dream come true because step by step after every new update I'm getting closer to using Obsidian as my IKE and IWE
> 
>     * I have to say that SC is changing my Obsidian-life :)

Nice to hear! 🙂 I can definitely agree that SC can help with _integrated knowledge environment_, at least in the _integrated_ part, as it can be used for "integrating" external tool applications more closely to Obsidian so that a user does not need to leave Obsidian so often when they want to perform external actions.

**Reply by FelipeRearden** [reply]
> Really interesting! I think I am bit by bit on my way on to do this kind of PKM things too.

Once we start, it's difficult to stop until you achieve something that you like :)

> Nice to hear! 🙂 I can definitely agree that SC can help with _integrated knowledge environment_, at least in the _integrated_ part, as it can be used for "integrating" external tool applications more closely to Obsidian so that a user does not need to leave Obsidian so often when they want to perform external actions.

Exactly! I must add that SC let me automate Obsidian with keyboard shortcut in a way that I can work faster (a good example is my `awk` SC)!
